build/wafutils.py

```diff
@@ -141,6 +141,31 @@ def check_c_header_cached(conf, **kw):
     return result
 
 
+def check_c_func_in_headers_cached(conf, **kw):
+    """
+    Check for @function_name in any of @header_name.
+    Use like a conf.check_cc() for a lib, but you can pass a list of
+    headers in @header_name.
+
+    This checks for the headers firs with check_c_header_cached(), then
+    if this succeeded calls conf.check_cc() on it.
+    """
+    headers = kw.get('header_name')
+    mandatory = kw.get('mandatory', True)
+
```

Comment:
Since both kwargs are passed explicitly to check_cc() below, we probably should .pop() them from the dict to not pass them twice to check_cc()